lms/djangoapps/course_api/blocks/transformers/video_urls.py

```diff
@@ -0,0 +1,67 @@
+"""
+Video block Transformer
+"""
+from __future__ import absolute_import
+
+import six
+from django.conf import settings
+
+from xmodule.video_module.video_utils import rewrite_video_url
+from openedx.core.djangoapps.content.block_structure.transformer import BlockStructureTransformer
+
+
+class VideoBlockURLTransformer(BlockStructureTransformer):
+    """
+    Transformer to re-write video urls for the mobile applications
+    to server content from edx-video.
+    """
+
+    @classmethod
+    def name(cls):
+        return "video_url"
+
+    WRITE_VERSION = 1
+    READ_VERSION = 1
+    STUDENT_VIEW_DATA = 'student_view_data'
+
+    def __init__(self):
+        self.cdn_url = getattr(settings, 'VIDEO_CDN_URL', {}).get('default', 'https://edx-video.net')
+        self.video_format_exceptions = ['youtube', 'fallback']
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        collect and store video block's student view data.
+        """
+        for block_key in block_structure.topological_traversal(
+            filter_func=lambda block_key: block_key.block_type == 'video',
+            yield_descendants_of_unyielded=True,
+        ):
+            xblock = block_structure.get_xblock(block_key)
+            block_structure.set_transformer_block_field(
+                block_key, cls, cls.STUDENT_VIEW_DATA, xblock.student_view_data()
```

Comment:
Currently, the way this is written, each video's `student_view_data` is being duplicated in the collection.  So we would need 2x the amount of storage for video data in our collection.  Instead, have this transformer assume that the `StudentViewTransformer` was called before this transformer.  And directly update the ``StudentViewTransformer`'s data in your `transform` method.

That is:
This `collect` method can be deleted altogether.
The `transform` method can then update the ``StudentViewTransformer`'s data directly.  This is Ok since these are "peer transformers" under the umbrella of the `BlocksAPITransformer`, once you move it there.

Reply:
I thought the same as the data would be accessible with `get_xblock_field`. But this will introduce the coupling which I didn't like adding.

Reply:
As long as the coupling remains within `BlocksAPITransformer` it is fine - since it would be localized within it.  `BlocksAPITransformer` is "umbrella" transformer and `StudentViewTransformer` and `VideoBlockURLTransformer` would be contained within as "sub-transformers".
There is already precedence of this as you will see the following comment within `BlocksAPITransformer`:
```
Note: BlockDepthTransformer must be executed before BlockNavigationTransformer.
```

Reply:
Right.